File: swift/GreenReadAR/Design/DesignTokens.swift
// DesignTokens.swift
// GreenRead AR — Design System Tokens
// PRD §5.2: Complete color palette + typography

import SwiftUI

extension Color {
    // PRD §5.2 Color Palette
    static let backgroundDeep = Color("BackgroundDeep")       // #0D1F14
    static let backgroundSurface = Color("BackgroundSurface") // #122A1A
    static let backgroundCard = Color("BackgroundCard")       // #183322
    static let primary = Color("Primary")                     // #1A7A4A
    static let primaryLight = Color("PrimaryLight")           // #22A55E
    static let downhill = Color("Downhill")                   // #3B82F6
    static let uphill = Color("Uphill")                       // #EF4444
    static let flat = Color("Flat")                           // #22C55E
    static let ball = Color("Ball")                           // #FFFFFF
    static let breakLine = Color("BreakLine")                 // #FACC15
    static let textPrimary = Color("TextPrimary")             // #F0FDF4
    static let textSecondary = Color("TextSecondary")         // #A7C4B8
    static let textDim = Color("TextDim")                     // #5A7D6C
    static let border = Color("Border")                       // rgba(26,122,74,0.2)
}

// Convenience extensions for hex colors used in RealityKit materials
extension UIColor {
    static let grdBackgroundDeep = UIColor(red: 0.051, green: 0.122, blue: 0.078, alpha: 1)
    static let grdPrimary = UIColor(red: 0.102, green: 0.478, blue: 0.290, alpha: 1)
    static let grdDownhill = UIColor(red: 0.231, green: 0.510, blue: 0.965, alpha: 1)
    static let grdUphill = UIColor(red: 0.937, green: 0.267, blue: 0.267, alpha: 1)
    static let grdFlat = UIColor(red: 0.133, green: 0.773, blue: 0.369, alpha: 1)
    static let grdBreakLine = UIColor(red: 0.980, green: 0.800, blue: 0.082, alpha: 1)
}

File: android/GreenReadAR/app/src/main/kotlin/com/greenreadar/ui/theme/DesignTokens.kt
// DesignTokens.kt
// GreenRead AR — Design System (Android)
// Port of DesignTokens.swift — PRD §5.2
//
// All 12 color tokens with exact hex values from PRD color palette.

package com.greenreadar.ui.theme

import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

/**
 * Design tokens for GreenRead AR.
 * Port of DesignTokens.swift — all colors per PRD §5.2.
 */
object DesignTokens {

    // MARK: - Colors (PRD §5.2)

    val BackgroundDeep = Color(0xFF0D1F14)
    val BackgroundSurface = Color(0xFF122A1A)
    val BackgroundCard = Color(0xFF183322)

    val Primary = Color(0xFF1A7A4A)
    val PrimaryLight = Color(0xFF22A55E)

    val Downhill = Color(0xFF3B82F6)     // Blue — downhill slopes
    val Uphill = Color(0xFFEF4444)       // Red — uphill slopes
    val Flat = Color(0xFF22C55E)         // Green — flat areas

    val Ball = Color(0xFFFFFFFF)         // White golf ball
    val BreakLine = Color(0xFFFACC15)    // Yellow break curve

    val TextPrimary = Color(0xFFF0FDF4)
    val TextSecondary = Color(0xFFA7C4B8)
    val TextDim = Color(0xFF5A7D6C)

    val Border = Color(0x331A7A4A)       // Primary with ~20% alpha

    // MARK: - Typography

    val FontTitle = FontFamily.Default
    val FontBody = FontFamily.Default

    val TitleSize = 28.sp
    val SubtitleSize = 18.sp
    val BodySize = 14.sp
    val CaptionSize = 12.sp
    val HudValueSize = 32.sp

    // MARK: - Spacing

    val PaddingSmall = 8.dp
    val PaddingMedium = 16.dp
    val PaddingLarge = 24.dp

    val CornerRadiusSmall = 8.dp
    val CornerRadiusMedium = 12.dp
    val CornerRadiusLarge = 20.dp

    // MARK: - Opacity

    const val GlassOpacity = 0.85f
    const val OverlayOpacity = 0.7f
}
